### DISCUSSION TITLE
How to propagate subworfklow's task status to the main workflow?

### DISCUSSION THREAD

**Discussion by MichalTkac** [Netflix/conductor#2498] [Q&A]
Hello,
we have a scenario where there are 3 workflows, let's call them A, B, C. Inside workflow A is called B workflow as sub-workflow with `optional` set to True then inside B workflow is workflow C called as sub-workflow. Inside workflow C are several dynamic fork tasks set with `optional`: True. If one of these tasks fails then the workflow is finished as COMPLETED but has some tasks with the status COMPLETED_WITH_ERRORS. 

We don't want that workflow to prevent us from stopping workflow A but we would like to see right away if there is even one task with another status than COMPLETED.
Is it possible to propagate this status from workflow C to workflow A? Could you suggest a way how this could be done, please?

Thank you,
Michal

**Comment by kishorebanala** [comment]
@MichalTkac There's no straight forward way to do this. One option I could think of is to use output variables, and propagate these variables all the way to workflow A. I.e the task with COMPLETED_WITH_ERRORS would have some output, which will be propagated to workflow output. This is then copied by subworkflow task in sub Workflow B and so on.

**Reply by MichalTkac** [reply]
@kishorebanala Thank you for your answer. I'm sorry for my late response. In that approach you came up with one won't be able to change the status of the main workflow A, right? The user would need to check if some specific value was parsed from bottom to top or maybe create a task that would check this parsed value. 
I think the only way how to control the status value somehow would be to use the TERMINATE task, right? 
Also, the task with COMPLETED_WITH_ERRORS doesn't know if it going to finish with errors or not.

**Comment by aravindanr** [comment]
If the SUB_WORKFLOW task is set as `optional` then Workflow A's status won't change.
